src/fold/mod.rs

```diff
@@ -365,7 +365,7 @@ macro_rules! copy_fold {
         impl ::fold::Fold for $t {
             type Result = Self;
             fn fold_with(&self,
-                         _folder: &mut ::fold::Folder,
+                         _folder: &mut dyn (::fold::Folder),
```

Comment:
@Manishearth -- well, I hit the edge case now =)

Reply:
Oh, that edge case is fine, it's the other way around. Will people ever be using `dyn::foo` as a path?

Reply:
> Will people ever be using dyn::foo as a path?

Who knows...